appengine/standard/xmpp_wikibot/wikibot.py

```diff
@@ -0,0 +1,207 @@
+    from sleekxmpp.util.misc_ops import setdefaultencoding
+    setdefaultencoding('utf8')
+else:
+    raw_input = input
+
+app = Flask(__name__)
+
+chat_client = None #this will be initialized when the wikibot is constructed
+
+@app.route('/send_message', methods=['GET'])
+def send_message():
+    try:
+        recipient = request.args.get('recipient')
+        message = request.args.get('message')
+    except KeyError as e:
+        logging.info("key error: {0}".format(e))
+
+    if chat_client and recipient and message:
+        chat_client.send_message(mto=recipient, mbody=message)
+        return "message sent to:" + recipient + " with body:" + message
+    else:
+        logging.info("chat client or recipient or message does not exist!")
+    return "message failed to send"
+
+def run_server():
+    app.run(threaded=False, use_reloader=False)
+
+class WikiBot(sleekxmpp.ClientXMPP):
+
+    """
+    A simple SleekXMPP bot that will take messages, look up their content
+    on wikipedia and provide a link to the page if it exists
+    """
+
+    def __init__(self, jid, password):
+        sleekxmpp.ClientXMPP.__init__(self, jid, password)
+
+        # The session_start event will be triggered when
+        # the bot establishes its connection with the server
+        # and the XML streams are ready for use. We want to
+        # listen for this event so that we we can initialize
+        # our roster.
+        self.add_event_handler("session_start", self.start)
+
+        # The message event is triggered whenever a message
+        # stanza is received. Be aware that that includes
+        # MUC messages and error messages.
+        self.add_event_handler("message", self.message)
+
+    def start(self, event):
+        """
+        Process the session_start event.
+        Typical actions for the session_start event are
+        requesting the roster and broadcasting an initial
+        presence stanza.
+        Arguments:
+            event -- An empty dictionary. The session_start
+                     event does not provide any additional
+                     data.
+        """
+        self.send_presence()
+        self.get_roster()
+
+    def message(self, msg):
+        """
+        Process incoming message stanzas. Be aware that this also
+        includes MUC messages and error messages. It is usually
+        a good idea to check the messages's type before processing
+        or sending replies. If the message is the appropriate type,
+        then the bot checks wikipedia to see if the message string
+        exists as a page on the site. If so, it sends this link back
+        to the sender in the reply.
+        Arguments:
+            msg -- The received message stanza. See the SleekXMPP documentation
+                   for stanza objects and the Message stanza to see
+                   how it may be used.
+        """
+        if msg['type'] in ('chat', 'normal'):
+            msg_body = "%(body)s" % msg
```

Comment:
Couldn't you just do `msg_body = msg['body']`? The string format seems gratuitous...